hail/python/hailtop/aiotools/tasks.py

```diff
@@ -21,9 +21,16 @@ def ensure_future(self, coroutine):
         if self._closed:
             raise TaskManagerClosedError
         t = asyncio.create_task(coroutine)
-        t.add_done_callback(lambda _: self.tasks.remove(t))
+        t.add_done_callback(self.on_task_done(t))
         self.tasks.add(t)
 
+    def on_task_done(self, t: asyncio.Task) -> Callable[[asyncio.Future], None]:
+        def callback(fut: asyncio.Future):
+            self.tasks.remove(t)
+            if e := fut.exception():
```

Comment:
I'm pretty sure if `fut` is cancelled this will raise a cancelled error. We should probably catch that, log it, and raise it?

Reply:
I don't know what happens when a callback raises.

Hmm. I suppose the callback has no reason to raise. There's no one awaiting us who should be cancelled.